packages/integration-objects/test/ValidateFieldsForRateTypeDesc.test.js

```diff
@@ -0,0 +1,110 @@
+/* Copyright (c) 2020 SAP SE or an SAP affiliate company. All rights reserved. */
+const { validateFieldsForExchangeRateTypeDescription } = require('../srv/validations/ValidateFieldsForRateTypeDesc');
+const RateTypeExtensionConstants = require('../srv/utils/RateTypeExtensionConstants');
+const ErrorStatuses = require('../srv/utils/ErrorStatuses');
+
+describe('validate fields of rate type description', function () {
+  it('does not throw any error', function () {
+    const reqObj = {
+      locale: 'en',
+      exchangeRateTypeDescription: 'sample'
+    };
+    const params = {
+      ID: 'id'
+    };
+    expect(validateFieldsForExchangeRateTypeDescription(reqObj, params)).toBeUndefined();
+  });
+
+  it('locale empty check', () => {
+    const reqObj = {
+      locale: null,
+      exchangeRateTypeDescription: 'sample'
+    };
+    const params = {
+      ID: 'id'
+    };
+    try {
+      validateFieldsForExchangeRateTypeDescription(reqObj, params);
+    } catch (err) {
+      expect(err.message).toBe('Provide a valid value for locale. The value must be 1 - 14 characters long.');
+    }
```

Comment:
If you catch the error, this test will also pass if there is no error thrown.
Please adjust this in all tests:
```suggestion
    expect(() => validateFieldsForExchangeRateTypeDescription(reqObj, params)).toThrowErrorMatchingInlineSnapshot('Provide a valid value for locale. The value must be 1 - 14 characters long.');
```

Reply:
@marikaner, @suhasraok and I tried all possible ways to get it working and finally fell back to the try-catch approach. Even this snapshot way fails.

Reply:
@vidhyakpai I fixed this one test to show you an example of how you can achieve it.
With the previous test, I was able to pass the test without throwing an error at all (I just returned true from the function and the test succeeded).

Please read the documentation on jest snapshots, it will help you to write tests. For the other tests, you can simply do something like this:
`expect(() => validateSomething(params)).toThrowErrorMatchingInlineSnapshot();`

The message will be filled in automatically, just check that it is in fact what you expect.